### PR TITLE
Support v2 Linter API

### LINKED ISSUE #121: Support v2 Linter API
In the [v2.3.0 release](https://github.com/steelbrain/linter/releases/tag/v2.3.0) of `linter` support for the v1 API was removed, breaking semver. This means that this provider is no longer supported by the current version of `linter`.

To get this provider working again it will need to be upgrade to the v2 API, a guide for accomplishing that can be found here: [Upgrading to Standard Linter v2](https://github.com/steelbrain/linter/blob/v2.3.0/docs/guides/upgrading-to-standard-linter-v2.md).

Until this is fixed in order to use this provider you will need to downgrade to the last version of `linter` that supports the v1 API, you can do this with `apm install linter@2.2.0`.

### PR DESCRIPTION
**Pull request by vzamanillo**
Fixes #121

### CODE REVIEW

**Comment by vzamanillo** [comment]
Any reason for using Go 1.8 with the stable channel and Go 1.9 with the beta channel?

**Comment by vzamanillo** [comment]
Tests are failing because of this https://github.com/golang/go/issues/2829, `TypeName.IsAlias` method was added in 1.9: https://golang.org/pkg/go/types/#TypeName.IsAlias

**Comment by Arcanemagus** [comment]
> Any reason for using Go 1.8 with the stable channel and Go 1.9 with the beta channel?

It's likely that when CI was configured Go 1.9 wasn't out yet.

File: package.json [hunk 1]
```diff
@@ -40,7 +40,7 @@
   },
   "dependencies": {
     "atom-linter": "^10.0.0",
-    "atom-package-deps": "^4.0.1"
+    "atom-package-deps": "^5.0.0"
```
[the remark this answers is not in the document]
Comment (vzamanillo) [reply package.json]: Didn't know, thank you!

File: lib/main.js [hunk 2]
```diff
@@ -9,11 +9,39 @@ let helpers = null;
...
 let extraOptions;
 
+const regex = /(.+):(\d+):(\d+):\s(.+)/g;
+
+const getMessagesFromStd = (std, textEditor, severity) => {
+  const messages = [];
+  if (std !== null || std !== '') {
+    let match = regex.exec(std);
+    while (match !== null) {
+      // Bump line number down 2 instead of 1 due to inserted extra line
```
[the remark this answers is not in the document]
Comment (vzamanillo) [reply lib/main.js]: Yep, sorry.

**Comment by vzamanillo** [comment]
Appveyor is fast as an old mule... :sleeping:

**Pull request merged**